tests/phpunit/tests/html-api/wpHtmlProcessorComments.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.7.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_WpHtmlProcessorComments extends WP_UnitTestCase {
+	/**
+	 * Ensures that different types of comments are processed correctly.
+	 *
+	 * @ticket 61530
+	 *
+	 * @dataProvider data_comments
+	 */
+	public function test_comment_processing( string $html, int $token_position, string $expected_token_name, string $expected_modifiable_text = '', string $expected_comment_type = null, string $expected_tag = null ) {
```

Comment:
this is a really long function signature and I think it could be split into two functions, the `$token_position` eliminated, and all of the optional values removed.

Apart from funky comments everything under test is a comment, which is in the test name, so we could then eliminate the `$expected_token_name` and simply assert that the first token is a `#comment`.

For the funky comment test it would be the same but for `#funky-comment`.